### ISSUE TITLE
Feature Suggestion: Configurable products in Bundle and/or Groups

### ISSUE THREAD

**Issue by intedyne** [magento/magento2#642]
It would be very helpful to add the ability of having configurable products as part of bundles and/or grouped products.

Use case: outfit builder. You have shirt, pants, and shoes. Each item has color and size. You want to sell the whole outfit, and allow customers to customize individual components.

Thank you.

**Comment by vpelipenko** [comment]
Internal ticket: MAGETWO-27900

**Comment by ontic** [comment]
I've achieved this this in Magento 1, allowing configurable & simple products to be bundled (even with required options). I really hope this get implemented in the Magento code base so I don't have to re-create this functionality again.

:+1:

**Comment by dustinnordquist** [comment]
This is functionality that many clients have requested.  Configurable Bundle 2.0 by Wizkunde is the best extension I've found to solve this need.  It could be used as a starting point.  I can also say the most requested improvements are 1) ability to set configurable product pricing based on the simple products, 2) tier pricing based on selected configurable/simple product quantities. Hope that helps.  See: http://www.magentocommerce.com/magento-connect/configurable-bundle-2-0-by-wizkunde.html

**Comment by claudiu-marginean** [comment]
:+1: If this functionality will be implemented in core Magento.

We had this request from one of our customers and used the extension "Grouped Options For Magento" from Brim.
http://ecommerce.brimllc.com/grouped-options.html

We had to custom code a lot of JS to make it work with "Pre-Select Colors Plus Swatches" crom CJM.
http://cjmcreativedesigns.com/magento-extensions/pre-select-colors-plus-swatches.html

Add on top of this an Ajax Add to cart and you have a good mix.

To support the level of prices that the customer wanted we had to implement the functionality @dustinnordquist mention in 1) and just found that this was already implemented in Magento 2 as part of issue #335

**Comment by urbancrew** [comment]
:+1:
Just like this:
http://demo.wizkunde.nl/configurable-bundle/my-bundle-product.html

**Comment by seansan** [comment]
+1

**Comment by PierAlex** [comment]
A customer is just asking for the ability to create a group of configurable products (t-shirts, pants, etc. having colour and size).
Will it be released nearly?

**Comment by seansan** [comment]
And it is o so necessary

This really is a missing feature!!! And in my opinion evne the 1.9x team should even make it backwards compatible and add it to 1.9x ....

See more detail here: http://magento.stackexchange.com/questions/113523/product-pack-of-1-base-configurable-product

http://magento.stackexchange.com/questions/70067/bundle-product-with-configurable-options

http://magento.stackexchange.com/questions/111799/magento-products-packs-assortiment-with-size-and-stock-control-like-3-pack-o

Best description: http://magento.stackexchange.com/questions/5002/multiple-configurable-products-as-a-bundle/5006?noredirect=1#comment159816_5006

The Wizkunde + Brim extension DO NOT support the exact request. Only simple request is: can we create a packaged (virtual) item out of an existing configurable product.

"Thx. In essence the only thing that needs to be done (to solve the question) is to create a stock multiplier. So if a pack consists of 3 sub products. Ordering 1 should result in qty = 3 for stock deduction. And even on invoice. So adding a multiplier observer should solve it everywhere and make it compatible with all other processes. Where would one add such a multiplier?"

**Comment by seansan** [comment]
@ontic : is your M1 code available somewhere to share? Would be interested

**Comment by urbancrew** [comment]
@seansan to add multiple configurable products as one package you need a extension like Wizkunde. That extension does exactly what the people here want to achieve.

**Comment by seansan** [comment]
I'd like to discuss that, because it does not. It creates a bundle of
_different_ products

We just need a configurable product, sold as a pack (t-shirts in sizes and
colours for example)

After some code discussions it is acctually quote simple (verbally ;) a
little more difficult in code

Packs of 1 product (in this case a pack/batch of 6 T-shirts) should be
shown and processed in the frontend as a product qty of 1. On product page,
in cart, in checkout, on the invoice

The only addition that is necessary is that on stock level settings a
"stock multiplier" should be created

So that when 1 pack is sold, qty =6 is deducted form the underlying simple
products

And when 1 pack is refunded and returned qty =6 is add to the underlying
simple products

etc etc

Maybe this should be recorded in a separate feature request ...

​Is it more clear now?​

**Comment by navarr** [comment]
This is definitely a missing needed feature.  I'm going to have to create some way to do it myself since there don't seem to be any M2 extensions to enable this functionality yet

**Comment by piotrekkaminski** [comment]
Thank you for your submission.

We recently made some changes to the way we process GitHub submissions to more quickly identify and respond to core code issues.

Feature Requests and Improvements should now be submitted to the new [Magento 2 Feature Requests and Improvements](https://community.magento.com/t5/Magento-2-Feature-Requests-and/idb-p/feature-requests) forum (see details [here](https://community.magento.com/t5/News-Announcements/Improvements-to-GitHub-Management/m-p/44572#M96)).

We are closing this GitHub ticket and have moved your request to the new [forum](https://community.magento.com/t5/Magento-2-Feature-Requests-and/Feature-Suggestion-Configurable-products-in-Bundle-and-or-Groups/idi-p/46200).

**Comment by jarnail104** [comment]
This functionality is also missing in M2. I get requirement to build Group product containing Configurable product but M2 dont provide this feature.

**Comment by RonXS** [comment]
We've been focussing our attention on allowing bundles to be a lot more useful over the course of 4 years. This ofcourse also included Magento 2 aswell. (Based on Configurables inside Bundles)

As seen here, its also possible in Magento 2:
https://m2.wizdemo.nl/balistico-2-0-with-rashguard.html

Extension: https://www.wizkunde.nl/magento-2-configurable-bundle.html

**Comment by jarnail104** [comment]
Ok. So should i post this in magento 2 feature request for adding configurable products in grouped? What you suggest

**Comment by CSnipper** [comment]
I've been searching this option in Magento, Presta, OpenCart, ZenCart, few Joomla scripts and only one script, SimpleCadyy has option, to embed multiple products witch options in one page. But it lacks all other features, like reports, advanced order management and others... It's ridiculous.

**Comment by ladle3000** [comment]
Also need this feature. Should not make magento merchants depend on 1 extension company for business critical feature.  Further, the flow of bundling simples can't handle much in m2 backend anyway.  So this would improve that by reducing the input.

**Comment by RonXS** [comment]
Maybe a little bit broad for this scope, but the bundle code as is, in conjunction with the EAV structure is a ticking timebomb in any shop. This has been the case since Magento introduced bundles, and when we started on our Configurable Bundle extension in Magento 1.6.x, this was our starting point.

When adding options to bundles, adding a significant amount of simple products would make the bundle pages so increadibly slow, it would be unusable. Ofcourse eventually when adding enough configurable products, would end up with the same problems.

If @magento-team is going to implement Configurables into bundles, there are a great deal of logics to take care of. This is probably why its still a todo on their list for a looooong time. 

A simple example is that regular bundled products dont show custom options from products in the bundle. So a fully functional product in default Magento cannot be bundled if the custom options are required. Just this, not taken Configurable products in account yet, only makes bundles useful in a very selective situation when the conditions are exactly right.

So to get a list of things to to, just for adding proper "sub product" support to a bundle or grouped, they need to work on:

- Rendering of super options in bundles/grouped products
- Rendering of cart contents
- Rendering of custom options for simple + configurable products
- Rendering of order, invoice, credit memo
- Backend functionality of bundle products (ofcourse)

Then the order process itself, from the cart on, keeping integrity in the buyRequest and such. 

We've got 6 years of experience with this, and our extension (i wont link it again, since people already referenced it), is a 6 year long journey of building and improving. It's one of the most complex Magento logics we've worked with and thats why you dont see much competition for our own extension.

Long story short: Integrating Configurable products in Bundles and Grouped products, properly, is a huge task at this point for Magento, which forces them to look at bundles from the ground up. And thats how it should be done, Bundles should be broken down into the concepts and built up from there.

**Comment by navarr** [comment]
@RonXS If you ever get tired of making money, you should contribute it to core 😆

**Comment by ladle3000** [comment]
@RonXS I've been emailing you and haven't heard back for a couple days.  Magento 2 has core bugs that importing doesn't work properly with bundles, sets them out of stock on front-end, there are multiple issues reported.  And the bundles I have already made from m1 are so big they can't be adjusted from the admin, it doesn't function. This is my urgency, as we've been on m2 a couple months and can't even update my bundles.

I agree it's a ticking time bomb, but I don't think it's as niche of a feature as they like to pretend.

**Comment by RonXS** [comment]
@navarr Haha yeah i've considered that and i've made some effort to work with the Magento team in the past on other manners (API functionality). But most of the core members are paid employees and many of the contributions are directly or indirectly used to market the Enterprise Edition. Thats basically building someone elses business. So thats not much of an incentive for me/us to do that ;-)

@snoroozi importing not just bundles but any entity has always been a problem, its one of the things that i have worked on in the past. If you get all the data to import properly, you'll end up with index problems. Hence there are so many other companies building importers like Magmi. We also use our own custom built import extension (not in our shop).
